api/v1/installation_types.go

```diff
@@ -141,6 +141,7 @@ const (
 	ComponentNameNode            ComponentName = "Node"
 	ComponentNameTypha           ComponentName = "Typha"
 	ComponentNameKubeControllers ComponentName = "KubeControllers"
+	ComponentNameECKOperator     ComponentName = "ECKOperator"
```

Comment:
You'll need to include below in the validation that ECKOperator is allowed.
I think also on Line 111 above you need to add a note that "Node, Typha, and KubeControllers" are supported.
Corresponding to that in `pkg/controller/installation/core_controller.go` should have validation that only those 3 are present on its componentResources.